src/doc/style-guide/src/items.md

```diff
@@ -295,8 +295,18 @@ Prefer to use single-letter names for generic parameters.
 
 These rules apply for `where` clauses on any item.
 
-If immediately following a closing bracket of any kind, write the keyword
-`where` on the same line, with a space before it.
+If a where clause is short, and appears on a short one-line function
+declaration with no body or a short associated type with no `=`, format it on
```

Comment:
Hopefully not being overly pedantic, but any objections to removing the "associated" word? I know over the years there's changes in terms of what parses and what doesn't but given rustfmt's operation in the pre-AST validation world I feel it may be safer to not have this qualifer

```suggestion
declaration with no body or a short type with no `=`, format it on
```

Reply:
No objection, though I would then want to write "or on a short type" to disambiguate and make sure it doesn't parse as "function declaration ... with ... a short type".

Reply:
i'd prefer that longer version if you're good with it :+1:

Reply:
```suggestion
declaration with no body or on a short type with no `=`, format it on
```